pip/pip-370.md

```diff
@@ -0,0 +1,48 @@
+# PIP-370: An optional flag: disable the replicators to automatically trigger topic creation
+
+# Background knowledge
+
+Users using Geo-Replication backup data across multi clusters, and Admin APIs related to Geo-Replication and internal replicators of brokers will trigger topics auto-creation between clusters.
+- For partitioned topics.
+  - After enabled namespace-level Geo-Replication: broker will create topics on the remote cluster automatically when calling `pulsar-admin topics create-partitioned-topic`. It does not depend on enabling `allowAutoTopicCreation`.
+  - When enabling topic-level Geo-Replication on a partitioned topic: broker will create topics on the remote cluster automatically. It does not depend on enabling `allowAutoTopicCreation`.
+- For non-partitioned topics and partitions of partitioned topics.
+  - The internal Geo-Replicator will trigger topics auto-creation for remote clusters. **(Highlight)** It depends on enabling `allowAutoTopicCreation`. In fact, this behavior is not related to Geo-Replication, it is the behavior of the internal producer of Geo Replicator,   
+
+# Motivation
+
+But if users want to maintain pulsar resources manually, there is no choice that prevent pulsar resources creation affects each other between clusters.
```

Comment:
Why would someone want to do this manually? It would be useful to explain a real use case here with proper argumentation.

Reply:
> Why would someone want to do this manually? It would be useful to explain a real use case here with proper argumentation.

Many users use a YAML file to define the pulsar resources on their cluster, such as bellow

```yaml
resources:
- public
  - default
    - tp-1
      - partitions: 3
    - tp-2
      - partitions: 0 
      - topic-policies: ... 
- tenant-1
  - ns-1
  - ...
```

There is a service that keeps reading the changes related to the file, compares the YAML file and actual pulsar resources, and calls pulsar rest APIs to keep actual pulsar resources matching the YAML file.

This solution is more clear for engineers than a Pulsar Admin UI.

Reply:
Thanks for the further explanation @poorbarcode.

With the yaml example, are you referring to the usage of [Pulsar Resources Operator](https://github.com/streamnative/pulsar-resources-operator)? 
Is the problem in that case that the auto-creation of the topic in the remote cluster causes some conflict with the usage of Pulsar Resources Operator in both clusters? Is that the problem that PIP-370 is intending to solve by allowing to disable the replicated topic creation in remote clusters?

Reply:
> Is the problem in that case that the auto-creation of the topic in the remote cluster causes some conflict with the usage of Pulsar Resources Operator in both clusters?
> Is that the problem that PIP-370 is intending to solve by allowing to disable the replicated topic creation in remote clusters?

**background**
- `enable topic level geo-replication` checks there are the same partitions between two clusters if the topic has already been created on the remote side.

Yes, the component such as the Resources Operator does not know the state of the remote cluster, the remote cluster may is not stable when it calls `enable topic level geo-replication`, so the Resources Operator can difficulty check what happened. After this feature, it can focus on the local cluster.

Reply:
>  the remote cluster may is not stable when it calls enable topic level geo-replication

clarifying question: what does it mean that a remote cluster "is not stable"?

Reply:
> clarifying question: what does it mean that a remote cluster "is not stable"?

For example:

**case-1**: Users deploying two clusters at the same time: when calling `pulsar-admin topics enable Geo-Replication` on the local cluster, maybe the remote cluster is starting.

**case-2**: the `cluster-2` is for `cluster-1`'s backup, so few resources are allocated to it, and it will be more unstable than the primary cluster.

**case-3**: the remote cluster are restarting